server/channels/api4/handlers.go

```diff
@@ -18,9 +18,13 @@ type Context = web.Context
 
 type handlerFunc func(*Context, http.ResponseWriter, *http.Request)
 
+const (
+	fileAPIKey = "fileAPI"
+)
+
 // APIHandler provides a handler for API endpoints which do not require the user to be logged in order for access to be
 // granted.
-func (api *API) APIHandler(h handlerFunc) http.Handler {
+func (api *API) APIHandler(h handlerFunc, opts ...interface{}) http.Handler {
```

Comment:
This seems hard to read because you have to go to several places before understanding what are the opts, how many are expected, in which order, etc...

I feel we could just add another wrapper function that receives the handler and return another handler that sets the needed stuff.

That way the handler would look something like...

```
api.BaseRoutes.Files.Handle("", withFileSizeLimit(api.APISessionRequired(uploadFileStream))).Methods("POST")
```

Reply:
Its not possible because the flags need to be set in web.Handler and all out handler like `APISessionRequired` etc return http.handler. I've thought about this a lot, trying about half a dozen ways, but this is the minimal one. I can update the type of opts instead of string to a new type to make it easier to see whats possible.

Reply:
Any other way to do this, at least what I thought of, requires changes in all API path declarations. I wanted to a avoid it and only specify something in API paths where we need something other than default.